### PR TITLE
Refactor doc flags (1 of n)

### PR DESCRIPTION
**Pull request by richsalz**
This is a WIP to show what the "move flags to common page" would look like.  #10115

### CODE REVIEW

File: doc/man1/openssl-dgst.pod [hunk 1]
```diff
@@ -190,6 +182,10 @@ specified.
 When used with the B<-engine> option, it specifies to also use
 engine B<id> for digest operations.
 
+=item B<-rand> I<file>, B<-writerand> I<file>
+
+See L<openssl(1)/Random state options> for more information.
+
```
Comment (levitte) [inline doc/man1/openssl-dgst.pod]: Why was this added?

File: doc/man1/openssl-srp.pod [hunk 2]
```diff
@@ -20,6 +20,8 @@ B<openssl srp>
 [B<-userinfo> I<text...>]
 [B<-passin> I<arg>]
 [B<-passout> I<arg>]
+[B<-rand> I<file...>]
+[B<-writerand> I<file...>]
```
Comment (levitte) [inline doc/man1/openssl-srp.pod]: Please don't add the ellipsis

File: util/find-doc-nits [hunk 3]
```diff
@@ -254,12 +257,14 @@ my $option_re =
             [[:alpha:]](?:[-_[:alnum:]]*?[[:alnum:]])?
        )/x;
 
+# Normal symbol form
+my $symbol_re = qr/[[:alpha:]_][_[:alnum:]]*?/;
+
```
Comment (levitte) [inline util/find-doc-nits]: I kept this near the function that uses it, why move it up?

File: doc/man1/openssl.pod [hunk 4]
```diff
@@ -545,6 +553,36 @@ Read the password from standard input.
...
+generally no longer necessary as OpenSSL will seed itself from the
+appropriate CPU flags, device files, and so on. These flags are still
+supported for special platforms or circumstances that might require them.
+
+It is generally an error to use the same seed file more than once and
+every use of B<-rand> should be paired with B<-writerand>.
+
+=over 4
+
+=item B<-rand> I<file>
```
Comment (levitte) [inline doc/man1/openssl.pod]: May I suggest plural, i.e `I<files>`, to indicate that it may be several files? When I look more closely at this option, that turned out to be a better way to say it than the ellipses we currently have for this option all over the place.

File: util/find-doc-nits [hunk 5]
```diff
@@ -803,30 +813,27 @@ sub checkflags {
...
-    }
-    if ( scalar @undocced > 0 ) {
-        foreach ( @undocced ) {
-            next if /-/; # Skip the -- end-of-flags marker
-            err("$doc: undocumented option -$_");
-        }
+    my @undocced = sort grep { !defined $docopts{$_} } keys %cmdopts;
+    foreach ( @undocced ) {
+        next if /-/; # Skip the -- end-of-flags marker
+        err("$doc: undocumented option -$_");
```
Comment (levitte) [inline util/find-doc-nits]: In particular, I very much like this refactoring (and the similar one a few lines down). Your perl-fu has notably increased.

Comment (richsalz) [reply util/find-doc-nits]: Thanks for the kind words.

File: doc/man1/openssl.pod (hunk 4 above)
Comment (richsalz) [reply doc/man1/openssl.pod]: that's an excellent idea.  will do.

File: doc/man1/openssl.pod [hunk 6]
```diff
@@ -545,6 +553,36 @@ Read the password from standard input.
...
+
+Prior to OpenSSL 3.0, it was common for applications to store information
+about the state of the random-number generator in a file that was loaded
+at startup and rewritten upon exit. On modern operating systems, this is
+generally no longer necessary as OpenSSL will seed itself from the
+appropriate CPU flags, device files, and so on. These flags are still
+supported for special platforms or circumstances that might require them.
+
+It is generally an error to use the same seed file more than once and
+every use of B<-rand> should be paired with B<-writerand>.
```
Comment (beldmit) [inline doc/man1/openssl.pod]: Hmm. Would it be better to enforce the presence of both (or none) options somewhere in apps.c?

File: doc/man1/openssl-dgst.pod (hunk 1 above)
Comment (richsalz) [reply doc/man1/openssl-dgst.pod]: Because it's in the code.  You can do ```./util/find-doc-nits -c | grep dgst``` for example.

File: doc/man1/openssl-srp.pod (hunk 2 above)
Comment (richsalz) [reply doc/man1/openssl-srp.pod]: cut/paste error, fixed.

File: util/find-doc-nits (hunk 3 above)
Comment (richsalz) [reply util/find-doc-nits]: i was just trying to group variables together.  i guess that's not perl style.  I will put it back.

File: doc/man1/openssl-dgst.pod (hunk 1 above)
Comment (levitte) [reply doc/man1/openssl-dgst.pod]: Let me ask in  a different way; why add a repeat when it's already present, 20 lines up?

Comment (richsalz) [reply doc/man1/openssl-dgst.pod]: oh, that.  cut/paste/paste error.  :)   fixed.

**Comment by levitte** [comment]
Unless you intend to do more substantial work, I see no reason to keep it as WIP
(when wondering, I always ask myself if I think it's complete enough and ready for final review/merge)

**Comment by richsalz** [comment]
I took it out of WIP.

**Comment by richsalz** [comment]
Rebased and pushed, ready for review.

File: doc/man1/openssl-dgst.pod [hunk 7]
```diff
@@ -25,6 +25,10 @@ B<openssl> B<dgst>|I<digest>
 [B<-hmac> I<key>]
 [B<-fips-fingerprint>]
 [B<-rand> I<files>]
+<<<<<<< c0724788a332f08833c4f9e4413ce9b9899d9260
+=======
+[B<-writerand> I<file>]
+>>>>>>> Address Richard's feedback
```
Comment (levitte) [inline doc/man1/openssl-dgst.pod]: Ehm...

File: doc/man1/openssl-dgst.pod [hunk 8]
```diff
@@ -184,7 +180,7 @@ specified.
 =item B<-engine_impl>
 
 When used with the B<-engine> option, it specifies to also use
-engine I<id> for digest operations.
+engine B<id> for digest operations.
```
Comment (levitte) [inline doc/man1/openssl-dgst.pod]: No

File: doc/man1/openssl-rand.pod [hunk 9]
```diff
@@ -19,12 +19,8 @@ I<num>
 
 =head1 DESCRIPTION
 
-This command outputs I<num> pseudo-random bytes after seeding
-the random number generator once.  As in other B<openssl> command
-line tools, PRNG seeding uses the file F<$HOME/.rnd> or F<.rnd>
-in addition to the files given in the B<-rand> option.  A new
-F<$HOME/.rnd> or F<.rnd> file will be written back if enough
-seeding was obtained from these sources.
+The B<rand> command outputs I<num> pseudo-random bytes after seeding
```
Comment (levitte) [inline doc/man1/openssl-rand.pod]: "This command..."

File: doc/man1/openssl-srp.pod (hunk 2 above)
Comment (levitte) [reply doc/man1/openssl-srp.pod]: Apparently not...

File: doc/man1/openssl-srp.pod [hunk 10]
```diff
@@ -20,7 +20,9 @@ B<openssl srp>
 [B<-userinfo> I<text>]
 [B<-passin> I<arg>]
 [B<-passout> I<arg>]
-[I<user> ...]
+[B<-rand> I<files>]
+[B<-writerand> I<file...>]
+[I<user...>]
```
Comment (levitte) [inline doc/man1/openssl-srp.pod]: Why did you move the ellipsis inside the markup?

File: doc/man1/openssl-dgst.pod (hunk 7 above)
Comment (richsalz) [reply doc/man1/openssl-dgst.pod]: heh. this was a painful rebase :)

File: doc/man1/openssl-dgst.pod (hunk 8 above)
Comment (richsalz) [reply doc/man1/openssl-dgst.pod]: undo.

File: doc/man1/openssl-srp.pod (hunk 10 above)
Comment (richsalz) [reply doc/man1/openssl-srp.pod]: habit.  fixed.

**Comment by richsalz** [comment]
All fixed, additional commit pushed.

File: doc/man1/openssl.pod [hunk 11]
```diff
@@ -554,6 +562,36 @@ Read the password from standard input.
...
+
+=over 4
+
+=item B<-rand> I<files>
+
+A file or files containing random data used to seed the random number
+generator.
+Multiple files can be specified separated by an OS-dependent character.
+The separator is B<;> for MS-Windows, B<,> for OpenVMS, and B<:> for
```
Comment (levitte) [inline doc/man1/openssl.pod]: Hmmm, not sure `B<` is right in this case...  `C<`?  Frankly, I'm not entirely sure.  @t8m, do you have an opinion?

File: doc/man1/openssl-crl.pod [hunk 12]
```diff
@@ -92,17 +94,9 @@ Output the lastUpdate field.
...
-Verify the signature on a CRL by looking up the issuing certificate in
-I<file>.
-
-=item B<-CApath> I<dir>
-
-Verify the signature on a CRL by looking up the issuing certificate in
-I<dir>. This directory must be a standard certificate directory: that
-is a hash of each subject name (using the L<openssl-x509(1)> B<-hash> option)
-should be linked to each certificate.
+See L<openssl(1)/Trusted certificate options> for more information.
```
Comment (levitte) [inline doc/man1/openssl-crl.pod]: Hey, isn't that section missing?

Comment (richsalz) [reply doc/man1/openssl-crl.pod]: oops, it was in another window.  And I also fixed the capitalization of the links.  new commit pushed.

File: doc/man1/openssl.pod [hunk 13]
```diff
@@ -554,6 +562,73 @@ Read the password from standard input.
...
 =back
 
+=head2 Trusted Certificate Options
+
+Part of validating a certificate includes verifying that the chain of CA's
+can be tied up to an existing trusted root.  The following options specify
+how to list the trusted roots, also known as trust anchors.  A collection
+of trusted roots is called a I<trust store>.
+
+Note that OpenSSL does not provider a default set of trust anchors.  Many
```
Comment (levitte) [inline doc/man1/openssl.pod]: We're all so used to say "provider" these days, aren't we?  I constantly catch myself making that exact same mistake

File: doc/man1/openssl-srp.pod [hunk 14]
```diff
@@ -65,10 +67,12 @@ adding or modifying a user.
 
 =item B<-passin> I<arg>, B<-passout> I<arg>
 
-The password source for the input and output file.
-For more information about the format of B<arg>
 see L<openssl(1)/Pass Phrase Options>.
```
Comment (levitte) [inline doc/man1/openssl-srp.pod]: Did you mean to do this in this PR?  Shouldn't that be in another `-passin` / `-passout` PR?  Or if you meant for it to be part of this PR, aren't there more places where this change should happen?

Either way, the remaining sentence should be this:

```
See L<openssl(1)/Pass Phrase Options> for more information.
```

File: doc/man1/openssl.pod (hunk 13 above)
Comment (richsalz) [reply doc/man1/openssl.pod]: Ha!  Fixed.

File: doc/man1/openssl-srp.pod (hunk 14 above)
Comment (richsalz) [reply doc/man1/openssl-srp.pod]: No, it was an editing error.  Restored.

**Comment by richsalz** [comment]
updated commit pushed.

File: doc/man1/openssl.pod (hunk 11 above)
Comment (t8m) [reply doc/man1/openssl.pod]: I would vote for `C<` as well.

File: doc/man1/openssl.pod [hunk 15]
```diff
@@ -554,6 +562,73 @@ Read the password from standard input.
...
+
+Use the specified directory as a list of trust certificates. That is,
+files should be named with the hash of the X.509 SubjectName of each
+certificate. This is so that the library can extract the IssuerName,
+hash it, and directly lookup the file to get the issuer certificate.
+See L<openssl-rehash(1)> for information on creating this type of directory.
+
+=item B<-no-CApath>
+
+Do not use load the default directory of trusted certificates.
```
Comment (t8m) [inline doc/man1/openssl.pod]: use or load?

File: doc/man1/openssl.pod [hunk 16]
```diff
@@ -554,6 +562,73 @@ Read the password from standard input.
...
+generator.
+Multiple files can be specified separated by an OS-dependent character.
+The separator is B<;> for MS-Windows, B<,> for OpenVMS, and B<:> for
+all others. Another way to specify multiple files is to repeat this flag
+with different filenames.
+
+=item B<-writerand> I<file>
+
+Writes the seed data to the specified specified I<file> upon exit.
+This can be used with on a subsequent command invocation.
```
Comment (t8m) [inline doc/man1/openssl.pod]: Specified is doubled and 'used with on' seems a little bit weird?

File: doc/man1/openssl.pod (hunk 15 above)
Comment (richsalz) [reply doc/man1/openssl.pod]: Obviously, I couldn't make up my mind :)  I decided "use" since nothing from the directory is actually loaded.

File: doc/man1/openssl.pod (hunk 16 above)
Comment (richsalz) [reply doc/man1/openssl.pod]: Rewrote:
```
Writes the seed data to the specified I<file> upon exit.
This file can be used in a subsequent command invocation.
```

**Comment by richsalz** [comment]
fixup commit pushed.

File: doc/man1/openssl.pod (hunk 11 above)
Comment (t8m) [reply doc/man1/openssl.pod]: @richsalz Please change this and I'll approve.

Comment (richsalz) [reply doc/man1/openssl.pod]: Amended commit pushed.  Is there guidance anywhere on when to use ```C<>``` vs ```B<>```?  We're not consistent, so it will be will-o-the-wisp, which is okay I guess.

I would prefer all commits are squashed when merging.

**Comment by richsalz** [comment]
squashed and rebased.  no changes made.  better comments on the commit.

**Comment by t8m** [comment]
My approval still holds.

**Comment by richsalz** [comment]
> My approval still holds.

You must be tired of saying that. :) Almost as much as I am waiting for the second approval :)

**Comment by richsalz** [comment]
This isn't "waiting for 2nd review" and can be merged now.

**Comment by t8m** [comment]
Unfortunately it is waiting for 2nd review from @openssl/omc Ping @levitte @mattcaswell ?

**Comment by t8m** [comment]
Merged [master a397aca435]
